dbms/src/Storages/StorageKafka.cpp

```diff
@@ -361,8 +416,18 @@ void StorageKafka::streamToViews()
 
     // Limit the number of batched messages to allow early cancellations
     const Settings & settings = context.getSettingsRef();
-    const size_t block_size = std::min(settings.max_block_size.value, BATCH_SIZE_MAX);
-    BlockInputStreamPtr in = std::make_shared<KafkaBlockInputStream>(*this, context, schema_name, block_size);
+    const size_t block_size = settings.max_block_size.value;
+
+    // Create a stream for each consumer and join them in a union stream
+    BlockInputStreams streams;
+    streams.reserve(num_consumers);
+    for (size_t i = 0; i < num_consumers; ++i)
+    {
+        auto consumer = claimConsumer(-1);
```

Comment:
-1 (with intention to implicitly convert to max long) looks weird.

Reply:
I agree 👍